### PR TITLE
[KS-135] Add WorkflowSpec

### PR DESCRIPTION
**Pull request by cedric-cordenier**

### CODE REVIEW

**Comment by cedric-cordenier** [comment]
Note: for these changes to be fully reflected in operator-ui, I need to merge the corresponding PR here: https://github.com/smartcontractkit/operator-ui/pull/74

File: core/services/job/models.go [hunk 1]
```diff
@@ -821,3 +823,11 @@ type LiquidityBalancerSpec struct {
 
 	LiquidityBalancerConfig string `toml:"liquidityBalancerConfig" db:"liquidity_balancer_config"`
 }
+
+type WorkflowSpec struct {
+	ID         int32     `toml:"-"`
+	WorkflowID string    `toml:"workflowId"`
```
Comment (bolekk) [inline core/services/job/models.go:829]: Workflow owner address as well?

File: core/store/migrate/migrations/0232_add_workflow_spec.sql [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+CREATE TABLE workflow_specs (
+    id              SERIAL PRIMARY KEY,
+    workflow_id     varchar(64) NOT NULL,
+    workflow        text NOT NULL,
+    created_at      timestamp with time zone NOT NULL,
+    updated_at      timestamp with time zone NOT NULL
+);
+
+INSERT INTO workflow_specs (workflow_id, workflow, created_at, updated_at)
+    VALUES ('15c631d295ef5e32deb99a10ee6804bc4af1385568f9b3363f6552ac6dbb2cef', '<DUMMY WORKFLOW>', NOW(), NOW());
```
Comment (bolekk) [inline core/store/migrate/migrations/0232_add_workflow_spec.sql:12]: why this?

**Review by bolekk — COMMENTED** [review]
Looks OK, I'm not sure if that's all that is needed ...

File: core/services/job/models.go (hunk 1 above)
Comment (cedric-cordenier) [reply core/services/job/models.go:829]: 👍

File: core/store/migrate/migrations/0232_add_workflow_spec.sql (hunk 2 above)
Comment (cedric-cordenier) [reply core/store/migrate/migrations/0232_add_workflow_spec.sql:12]: This was needed to be able to deploy this PR to staging -- without it we fail the constraint. I was hesitant to remove this after deployment on the off chance that the migration will fail for others (although I don't really see how!)

File: core/services/job/models.go (hunk 1 above)
Comment (cedric-cordenier) [reply core/services/job/models.go]: Alright, added :)

File: core/store/migrate/migrations/0232_add_workflow_spec.sql (hunk 2 above)
Comment (cedric-cordenier) [reply core/store/migrate/migrations/0232_add_workflow_spec.sql]: Having thought about this more, I'd like to keep this in out of an abundance of caution -- I've added the hardcoded workflow as the default and added some default values for the WorkflowOwner / Workflow ID.

**Review by bolekk — DISMISSED** [review]

**Review by bolekk — APPROVED** [review]

File: core/store/migrate/migrations/0232_add_workflow_spec.sql [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+CREATE TABLE workflow_specs (
+    id              SERIAL PRIMARY KEY,
+    workflow_id     varchar(64) NOT NULL,
+    workflow        text NOT NULL,
+    workflow_owner  varchar(40) NOT NULL,
+    created_at      timestamp with time zone NOT NULL,
+    updated_at      timestamp with time zone NOT NULL
+);
+
+INSERT INTO workflow_specs (workflow_id, workflow, workflow_owner, created_at, updated_at)
```
Comment (bolekk) [inline core/store/migrate/migrations/0232_add_workflow_spec.sql:12]: I'm now realizing that this should't be here, let's remove it.

**Pull request merged**